poc/trait_list_object.py

```diff
@@ -0,0 +1,895 @@
+
+        Notes
+        -----
+        Notification:
+            index : slice
+                Slice ranging from current_length to new_length.
+            added : list
+                The  new values that were added.
+            removed : list
+                The empty list [].
+
+        """
+        self.extend(other)
+        return self
+
+    def __imul__(self, count):
+        """ Implements the self *= value operator.
+
+        Parameters
+        ----------
+        count : int
+            Number of times to duplicate the list.
+
+        Notes
+        -----
+        Notification:
+            index : slice
+                Slice ranging from current_length to new_length.
+            added : list
+                The  new values that were added.
+            removed : list
+                The empty list [].
+
+        """
+        if count > 1:
+            self.extend(self * (count - 1))
+        elif count == 0:
+            self[:] = []
+        return self
+
+    # ------------------------------------------------------------------------
+    # Private interface
+    # ------------------------------------------------------------------------
+
+    def _get_removed(self, index):
+        """ Compute removed values given index. """
+        try:
+            if not isinstance(index, slice):
+                return [self[index]]
+            else:
+                return self[index]
+        except IndexError:
+            return []
+
+    def _normalize(self, index):
+        if isinstance(index, slice):
+            return self._normalize_slice(index)
+        else:
+            return self._normalize_index(index)
+
+    def _normalize_index(self, index):
+        """ Normalize integer index to range 0 to len (inclusive). """
+        index = operator.index(index)
+        if index < 0:
+            return max(0, len(self) + index)
+        else:
+            return min(len(self), index)
+
+    def _normalize_slice(self, index):
+        """ Normalize slice start and stop to range 0 to len (inclusive). """
+
+        # Do not normalize if step is negative.
+        if index.step is not None and index.step < 0:
+            return index
+
+        if index.step is None or index.step > 0:
+            if index.start is not None:
+                start = self._normalize_index(index.start)
+            else:
+                start = 0
+            if index.stop is not None:
+                stop = self._normalize_index(index.stop)
+            else:
+                stop = len(self)
+        else:
+            if index.start is not None:
+                start = self._normalize_index(index.start)
+            else:
+                start = len(self)
+            if index.stop is not None:
+                stop = self._normalize_index(index.stop)
+            else:
+                stop = 0
+
+        return slice(start, stop, index.step)
+
+    def _should_notify(self, removed, added):
+        return True
```

Comment:
This is changed from #912 (equivalent to removing all `if _should_notify` in other places).